### PR TITLE
Feature/pipeline improvements

### PR DESCRIPTION
**Pull request by codeflyer**
In this PR: #132 

* Add support to `privileged` attribute in pipeline creation
* Fixed a naming issues in cicd permissions creation
* Removed the required repository url for `codecommit` repository type

### CODE REVIEW

**Review by frikille — APPROVED** [review]
👍

File: docs/cicd/README.md [hunk 1]
```diff
@@ -81,12 +81,14 @@ Then, modify your `cicd` section in the config file (`default.json`) and create/
...
-        "requireManualApproval": false
+        "requireManualApproval": false,
+        "privileged": true
       },
       {
         "target": "production",
         "withDomain": false,
-        "requireManualApproval": true
+        "requireManualApproval": true,
+        "privileged": true
```
Comment (mikegwhit) [inline docs/cicd/README.md:91]: Just interested to know more what the privileged flag does?

File: docs/config/README.md [hunk 2]
```diff
@@ -116,6 +118,8 @@ Stage is described by 3 properties:
     * `target` - Name of the account used as a target account for the application deployment (e.g. staging).
     * `withDomain` - A boolean that specifies if the application supports a custom domain. __NB domain usage to follow in upcoming releases.__.
     * `requireManualApproval` - A boolean that specifies if manual approval is needed in the pipeline before continuing deployment.
+    * `privileged` - A boolean that specifies if the pipeline should support the `privileged` environment value.
```
Comment (mikegwhit) [inline docs/config/README.md:121]: Still interested in what this achieves

File: src/cdk/constructs/cicd/deployment-permissions/index.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ export class DeploymentPermissions extends Stack {
   public role: Role
 
   constructor (parent: Construct, props: DeploymentPermissionsProps = { env: 'dev' }) {
-    const name = getBaseStackName('CICDPermissions')
+    const name = `${MiraConfig.getBaseStackName('CICDPermissions')}-${props.env}`
```
Comment (mikegwhit) [inline src/cdk/constructs/cicd/deployment-permissions/index.ts:15]: Just noting from yesterday's call, this is an area where I had mentioned we can abstract away naming.

File: src/cdk/constructs/cicd/index.ts [hunk 4]
```diff
@@ -137,7 +137,10 @@ export class Cicd extends Stack {
         description: 'Project repository'
       })
       new CfnOutput(this, 'RepositoryName', {
-        value: MiraConfig.calculateRepositoryName()
+        value: repository.repositoryName
```
Comment (mikegwhit) [inline src/cdk/constructs/cicd/index.ts:140]: Nice

File: src/config/mira-config.ts [hunk 5]
```diff
@@ -194,14 +196,19 @@ class MiraConfigClass {
 
   public getCICDConfig (): CicdConfig {
     const output = this.getFullCiProps(CONFIG_KEYS.CICD)
-    const {
-      name,
-      owner
-    } = parseGitUrl(output.repositoryUrl)
+    const githubValues: { repositoryName?: string, repositoryOwner?: string } = {}
+    if (output.provider !== 'codecommit') {
```
Comment (mikegwhit) [inline src/config/mira-config.ts:200]: This might be painful to test considering, but just wondering if this was tested w GitHub vs. CodeCommit?

**Review by mikegwhit — APPROVED** [review]

File: docs/cicd/README.md (hunk 1 above)
Comment (codeflyer) [reply docs/cicd/README.md:91]: The `privileged` should be set to true to allow docker stuff in the pipeline.

Without it this command won't work:

```
docker build -t ${ECR_REPOSITORY_NAME} . 
```

It sets this value inAWS pipeline:
Screenshot 2020-10-13 at 16 14 32

File: src/cdk/constructs/cicd/deployment-permissions/index.ts (hunk 3 above)
Comment (codeflyer) [reply src/cdk/constructs/cicd/deployment-permissions/index.ts:15]: Yep, we probably need an improvement in the Config/Object class. And a well defined docs with rules on how to retrieve values from the config.json file

File: src/config/mira-config.ts (hunk 5 above)
Comment (codeflyer) [reply src/config/mira-config.ts:200]: No, the test with github wasn't done. But I think it worth to do be done.

**Pull request merged**